tests/contrib/grpc_aio/test_grpc_aio.py

```diff
@@ -1005,3 +1035,130 @@ async def test_client_streaming(server_info, tracer):
     out, err, status, _ = ddtrace_run_python_code_in_subprocess(code, env=env)
     assert status == 0, (err.decode(), out.decode())
     assert err == b"", err.decode()
+
+
+class StreamInterceptor(grpc.aio.UnaryStreamClientInterceptor):
+    async def intercept_unary_stream(self, continuation, call_details, request):
+        response_iterator = await continuation(call_details, request)
+        return response_iterator
+
+
+async def run_streaming_example(server_info, use_generator=False):
+    i = 0
+    async with grpc.aio.insecure_channel(server_info.target, interceptors=[StreamInterceptor()]) as channel:
+        stub = MultiGreeterStub(channel)
+
+        # Read from an async generator
+        if use_generator:
+            async for response in stub.sayHello(HelloRequestStream(name="you")):
+                assert response.message == "Hello number {}, you!".format(i)
+                i += 1
+
+        # Direct read from the stub
+        else:
+            hello_stream = stub.sayHello(HelloRequestStream(name="will"))
+            while True:
+                response = await hello_stream.read()
+                if response == grpc.aio.EOF:
+                    break
+                assert response.message == "Hello number {}, will!".format(i)
+                i += 1
+
+
+@pytest.mark.asyncio
+@pytest.mark.skip("Bug/error from grpc when adding an async streaming client interceptor throws StopAsyncIteration")
+@pytest.mark.parametrize("async_server_info", [_CoroHelloServicer()], indirect=True)
+async def test_async_streaming_direct_read(async_server_info, tracer):
+    await run_streaming_example(async_server_info)
+
+    spans = _get_spans(tracer)
+    assert len(spans) == 2
+    client_span, server_span = spans
+
+    # No error because cancelled after execution
+    _check_client_span(client_span, "grpc-aio-client", "SayHelloRepeatedly", "bidi_streaming")
+    _check_server_span(server_span, "grpc-aio-server", "SayHelloRepeatedly", "bidi_streaming")
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("async_server_info", [_CoroHelloServicer()], indirect=True)
+async def test_async_streaming_generator(async_server_info, tracer):
+    await run_streaming_example(async_server_info, use_generator=True)
+
+    spans = _get_spans(tracer)
+    assert len(spans) == 2
+    client_span, server_span = spans
+
+    # No error because cancelled after execution
+    _check_client_span(
+        client_span, "grpc-aio-client", "sayHello", "server_streaming", "hellostreamingworld.MultiGreeter"
+    )
+    _check_server_span(
+        server_span, "grpc-aio-server", "sayHello", "server_streaming", "hellostreamingworld.MultiGreeter"
+    )
+
+
+repr_test_cases = [
```

Comment:
good! perhaps useful to also cases where the right-hand side of each equal is missing

Reply:
ok done